src/test/java/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt

```diff
@@ -325,4 +325,76 @@ class PythonFrontendTest : BaseTest() {
         assertEquals(TypeParser.createFrom("int", false), elseExpr?.type)
         assertEquals(42, (elseExpr?.value as? Long)?.toInt())
     }
+
+    @Test
+    fun testSelf() {
+        val topLevel = Path.of("src", "test", "resources", "python")
+        val tu =
+            TestUtils.analyzeAndGetFirstTU(
+                listOf(topLevel.resolve("class_self.py").toFile()),
+                topLevel,
+                true
+            ) {
+                it.registerLanguage(
+                    PythonLanguageFrontend::class.java,
+                    PythonLanguageFrontend.PY_EXTENSIONS
+                )
+            }
+
+        assertNotNull(tu)
+
+        val p =
+            tu.getDeclarationsByName("class_self", NamespaceDeclaration::class.java)
+                .iterator()
+                .next()
+
+        val Foo = p.getDeclarationsByName("Foo", RecordDeclaration::class.java).iterator().next()
+
+        assertNotNull(Foo)
+        assertEquals("Foo", Foo.name)
+
+        assertEquals(1, Foo.fields.size)
+        val somevar = Foo.fields[0] as? FieldDeclaration
+        assertEquals("somevar", somevar?.name)
+        assertEquals(TypeParser.createFrom("int", false), somevar?.type)
+
+        assertEquals(2, Foo.methods.size)
+        val bar = Foo.methods[0] as? MethodDeclaration
+        val foo = Foo.methods[1] as? MethodDeclaration
+        assertEquals("bar", bar?.name)
+        assertEquals(Foo, bar?.recordDeclaration)
+        assertEquals("foo", foo?.name)
+        assertEquals(Foo, foo?.recordDeclaration)
+
+        val recv = bar?.receiver as? VariableDeclaration
+        assertEquals("self", recv?.name)
+        assertEquals(TypeParser.createFrom("Foo", false), recv?.type)
+
+        assertEquals(1, bar?.parameters?.size)
+        val i = bar?.parameters?.get(0)
+        assertEquals("i", i?.name)
+
+        val body = bar?.body as? CompoundStatement
+        val assign = body?.statements?.get(0) as? BinaryOperator
```

Comment:
a quick `assertNotNull(body)` /  `assertNotNull(assign)` would help to reduce the amount of `?` operators ;)